# Meson wrapdb's meson.build file doesn't handle installing the library
# and find imgui system library, we patch that here

diff --git a/meson.build b/implot-0.16-meson.build
index 3343dd4ad..804f84227 100644
--- a/meson.build
+++ b/implot-0.16-meson.build
@@ -5,8 +5,7 @@
   version : '0.16',
 )

-imgui_sp = subproject('imgui')
-imgui_dep = imgui_sp.get_variable('imgui_dep')
+imgui_dep=dependency('imgui', required : true)

 implot_inc = include_directories('.')
 implot_src = files(
@@ -17,7 +16,19 @@
 implot_lib = library(
   'implot',
   implot_src,
-  dependencies: imgui_dep
+  dependencies: imgui_dep,
+  version: meson.project_version(),
+  install: true
+)
+
+pkg_mod = import('pkgconfig')
+pkg_mod.generate(implot_lib,
+  description : 'Immediate Mode Plotting'
+)
+
+install_headers(
+    'implot.h',
+    'implot_internal.h'
 )

 implot_dep = declare_dependency(
diff --git a/implot.h b/implot.h
index 3054331..856eb0c 100644
--- a/implot.h
+++ b/implot.h
@@ -45,7 +45,7 @@
 // [SECTION] Obsolete API
 
 #pragma once
-#include "imgui.h"
+#include <imgui/imgui.h>
 
 //-----------------------------------------------------------------------------
 // [SECTION] Macros and Defines
diff --git a/implot_internal.h b/implot_internal.h
index cd05478..a189fcf 100644
--- a/implot_internal.h
+++ b/implot_internal.h
@@ -32,7 +32,7 @@
 #pragma once

 #include <time.h>
-#include "imgui_internal.h"
+#include <imgui/imgui_internal.h>

 #ifndef IMPLOT_VERSION
 #error Must include implot.h before implot_internal.h
